jetty-ee9/jetty-ee9-websocket/jetty-ee9-websocket-jetty-common/src/test/java/org/eclipse/jetty/ee9/websocket/common/MessageOutputStreamTest.java

```diff
@@ -28,29 +32,49 @@
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.endsWith;
 import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
 
 public class MessageOutputStreamTest
 {
     private static final Logger LOG = LoggerFactory.getLogger(MessageOutputStreamTest.class);
     private static final int OUTPUT_BUFFER_SIZE = 4096;
 
-    public TestableLeakTrackingBufferPool bufferPool = new TestableLeakTrackingBufferPool("Test");
-
-    @AfterEach
-    public void afterEach()
-    {
-        bufferPool.assertNoLeaks();
-    }
-
+    private final AtomicInteger leaks = new AtomicInteger();
+    private RetainableByteBufferPool bufferPool;
     private OutgoingMessageCapture sessionCapture;
 
     @BeforeEach
-    public void setupTest() throws Exception
+    public void beforeEach()
     {
+        bufferPool = new ArrayRetainableByteBufferPool()
```

Comment:
Is this a leak tracking `RetainableByteBufferPool` implementation? why delete `TestableLeakTrackingBufferPool` then.

Reply:
Because it was using the other pool class.

Reply:
But why not just update `TestableLeakTrackingBufferPool` to be a `RetainableByteBufferPool` then other tests can potentially use it.